server/protocols/jmap-integration-testing/jmap-integration-testing-common/src/test/java/org/apache/james/jmap/methods/integration/cucumber/UserStepdefs.java

```diff
@@ -19,36 +19,97 @@
 
 package org.apache.james.jmap.methods.integration.cucumber;
 
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
 import javax.inject.Inject;
 
 import org.apache.james.jmap.JmapAuthentication;
 import org.apache.james.jmap.api.access.AccessToken;
+import org.apache.james.mailbox.model.MailboxConstants;
+
+import com.github.fge.lambdas.Throwing;
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.common.hash.Hashing;
 
+import cucumber.api.PendingException;
 import cucumber.api.java.en.Given;
 import cucumber.runtime.java.guice.ScenarioScoped;
 
 @ScenarioScoped
 public class UserStepdefs {
 
     private final MainStepdefs mainStepdefs;
-    protected String username;
-    protected AccessToken accessToken;
-
+    
+    protected Map<String, String> passwordByUsers;
+    protected Set<String> domains;
+    protected Map<String, AccessToken> tokenByUser;
+    protected String lastConnectedUser;
+    
     @Inject
     private UserStepdefs(MainStepdefs mainStepdefs) {
         this.mainStepdefs = mainStepdefs;
+        this.domains = new HashSet<>();
+        this.passwordByUsers = new HashMap<>();
+        this.tokenByUser = new HashMap<>();
     }
 
     @Given("^a domain named \"([^\"]*)\"$")
     public void createDomain(String domain) throws Throwable {
         mainStepdefs.jmapServer.serverProbe().addDomain(domain);
+        domains.add(domain);
     }
 
-    @Given("^a current user with username \"([^\"]*)\" and password \"([^\"]*)\"$")
-    public void createUserWithPasswordAndAuthenticate(String username, String password) throws Throwable {
-        this.username = username;
+    @Given("^some users (.*)$")
```

Comment:
already seen but : 
x the default delimiter is "\s," anyway
x it won't handle quoted string anyway and thus add nothing